### ISSUE TITLE
small genome size, but low ctg N50

### ISSUE THREAD

**Issue by whc2** [chhylp123/hifiasm#143]
Hello,

The plant genome has very low heterozygosity (the following fig) and genome size is ~600 Mb.
![image](https://user-images.githubusercontent.com/26691382/123034843-e6b92c00-d41c-11eb-87b5-1c3262dc190f.png)

The input Hifi data has 2 M reads, 27 Gb. I used `-l0` parameter and generated 3,130 ctgs, totaling 519 Mb (N50 569 Kb). Attachment is the log file. Then I tuned parameters `-z 20 -D 20 -r 15 -l1` and got 2,723 ctgs, totaling 524 Mb (N50 850 Kb). All numbers is about p_ctg. I don't know if these settings are reasonable? How can I improve N50 further?

I think this N50 is too low for such a small genome. Because N50 is 75 Mb for another ~1.8 Gb plant genome. The GC percent is 32% for the small genome, 36.7% for the later. Does GC percent difference is the main reason for the low N50 of the small genome.

Please give me some hints about how to improve this assembly. 
[hifiasm.log](https://github.com/chhylp123/hifiasm/files/6698916/hifiasm.log)

Thank you very much!

**Comment by chhylp123** [comment]
Well, I worry it is caused by data quality (see: https://github.com/chhylp123/hifiasm/issues/93#issuecomment-863916776 and https://github.com/chhylp123/hifiasm/issues/93#issuecomment-864085158). Have you also tried to assemble by HiCanu or IPA?

**Comment by whc2** [comment]
HiCanu generated > 6K ctgs, totaling ~600 Mb (N50 409 Kb). I haven't try IPA. Is it possible that this small genome have much too large repeat (> 10 Kb), so reads can not pass it.

**Comment by chhylp123** [comment]
It is possible, but I guess it shouldn't. Does it have CLR assembly?

**Comment by whc2** [comment]
We have ultra long Nanopore data (~110X). There are some assemblies: Flye (1 K ctgs, 436 Mb, N50 1.8 Mb); Shasta (13 K ctgs, 396 Mb, N50 1 Mb); Canu correction + Wtdbg2 (1.5 K ctgs, 390 Mb, N50 3.5 Mb). So, the assembly of ONT is smaller and Busco complete 95% vs. 97% of Hifi.

**Comment by whc2** [comment]
We have ultra long Nanopore data (~110X). There are some assemblies: Flye (1 K ctgs, 436 Mb, N50 1.8 Mb); Shasta (13 K ctgs, 396 Mb, N50 1 Mb); Canu correction + Wtdbg2 (1.5 K ctgs, 390 Mb, N50 3.5 Mb). So, the assembly of ONT is smaller and Busco complete 95% vs. 97% of Hifi.

**Comment by whc2** [comment]
We have ultra long Nanopore data (~110X). There are some assemblies: Flye (1 K ctgs, 436 Mb, N50 1.8 Mb); Shasta (13 K ctgs, 396 Mb, N50 1 Mb); Canu correction + Wtdbg2 (1.5 K ctgs, 390 Mb, N50 3.5 Mb). So, the assembly of ONT is smaller and Busco complete 95% vs. 97% of Hifi.

**Comment by chhylp123** [comment]
Then I have no idea. Looks like the sample itself is hard, or there is no enough DNA? Could you please check why the contigs are fragmented? Probably you can map HiFi reads back to assembly, and have a look at the alignment around breakpoints.

**Comment by whc2** [comment]
Yes, I will do it. Thank you.

**Comment by HenrivdGeest** [comment]
I am also wondering what the read lengths are of your hifi data. (Or ont)
Plus your coverage peaks at 60x, you could sample on 30x longest reads for
example, or if you have long reads, take the 30x shortest ( shorter = more
accurate. )
Plus what does genomescope say, with kmer of 128 or even 256? With this
coverage it should not be a problem to estimate genome size even with a
kmer of 128.
Overall, interesting plant you have there :)

**Comment by whc2** [comment]
The reads N50s are >13 Kb and > 48 Kb for hifi and ont. I used the short part of hifi data and didn't have much assembly length difference. The genome size is estimated by illumina data. The assembly size is shorter, because complex repeats can't be produced completely.

I checked the assembly graph and fond this genome has very complex paths in r_utg.gfa. Large parts of utgs have no connection.
![594E8D885D62F3B6449F2575292CCF80](https://user-images.githubusercontent.com/26691382/123527807-2e89cd00-d715-11eb-80d2-f62d3b4df891.png)
 
Most of them are like this:
![D9A1D47990BC772723B12597AF73A68B](https://user-images.githubusercontent.com/26691382/123527823-482b1480-d715-11eb-8eac-e9005db712b0.png)

And even this:
![D65F2B450BE84016D8BA0C6E0210C76B](https://user-images.githubusercontent.com/26691382/123527829-5842f400-d715-11eb-9fad-0f7601d568fe.png)



I did mapped hifi reads to the assembly, but not sure how to `have a look at the alignment around breakpoints.`
